### DISCUSSION TITLE
option for constant bit rate

### DISCUSSION THREAD

**Discussion by kaelsonofkrypto** [nagadomi/nunif#88] [General]
Please add the option for Being able to add a custom constant bit rate for H.264. Thanks for all your efforts and hardware. and hard work.

**Comment by nagadomi** [comment]
This can be supported, but I would prefer not to do it. Because I think it makes GUI and code more complicated on less important issue.
Do you have a serious issue with CRF (Constant Rate Factor)?

**Reply by kaelsonofkrypto** [reply]
Thanks for the fast reply, the main reason was because i noticed banding even when i lowered to 16, if constant bit rate will cause issues then no worries, but would it be possible to give lower crf options like 8 that wont compress as much

**Reply by nagadomi** [reply]
I have made CRF editable. You can specify any value.
Also, if you specify `grain` for `Tune`, it gives higher bit rate for the same CRF.

**Reply by kaelsonofkrypto** [reply]
thanks so much

**Reply by kaelsonofkrypto** [reply]
hi, can you use two 3090s in the same pc with this program? i was wondering because i actually have two 3090s but one in each pc but was wondering if i put two in the same pc would i be able to use the combination of two to get to higher depth resolutions, because it seems like with one 3090 my cut off is around 1175 depth resolution. I can totally see getting to higher depth resolutions begins clearing up artifacts a lot.

**Reply by nagadomi** [reply]
A single data cannot be processed with multiple GPUs.

Multi-GPU feature is supported only for Data Parallel.
Data Parallel mean that when processing multiple data, each data is divided into multiple GPUs.
For example, when batch size=8, 4 batches can be processed in parallel on two GPUs.
(Also, this feature is currently only enabled for ZoeDepth. I have not tested DepthAnything on multi-GPU environment and am not sure if it works, so it is disabled.)

To reduce VRAM usage for single data, an easy way is to reduce the floating-point format size.
FP16 (16-bit float) can reduce VRAM usage to half compared to FP32.
With CPU, it is not limited by VRAM size.

Regarding artifacts caused by image warping (ghost artifact), I am currently re-training the row_flow model with recent changes, it may be improved a little.

**Reply by kaelsonofkrypto** [reply]
thanks so much for the detailed answer,  so if i select fp16, i just want to make sure that doing both fp16 and tta together will not reduce the quality of the 3d?

 and glad you are retraining it to hopefully result in fewer artifacts. that's so awesome of you,

**Reply by nagadomi** [reply]
fp16 is not completely free of degradation.
I tried to compare the depth between fp16 and fp32 for some videos using --export. The average PSNR was 56.
PSNR 56 is practically the same result. A person of ordinary ability cannot see the difference.
With fp16, the processing time and VRAM usage will be roughly 1/2. There are more advantages than disadvantages.

TTA-related image processing is always performed with fp32, so fp16 option has no effect. 
fp16 is used only for model execution. All other image processing is performed with fp32.
(affects only the source depth image)

**Reply by kaelsonofkrypto** [reply]
thanks. really appreciate the clarification.

**Comment by kaelsonofkrypto** [comment]
Hey , have a quick question , how do you use the resume function on iw3 if u accidentally didn't make enough space and a conversion crashed due to no space , thanks in advance

**Reply by nagadomi** [reply]
Currently resume is not supported.
For resume to work perfectly, frames and depths must be saved as image files and encoded at the end.
Then it is more efficient to simply run it twice, only when trouble occurs.

If you really want that feature, I will add resume option to --export.
In this case, the basic workflow is
1. export (save frame images and depth images, audio)
2. generate video (encode)
3. (delete temporary image files)

**Reply by kaelsonofkrypto** [reply]
yes please that would be such a great help if added, sometimes i experience outages in area because we get peak demand and we get outages, sometime in summer they even purposely do what they call rolling black outs in our area to conserve power. the bad part of this is sometimes people are without power when they do these rolling blackouts, so a resume on export function would definitely be helpful. if you ever find a way to do it on video to that would be awesome, but i would totally be appreciative of even just the function on export being an option.

Thanks for you taking the time to respond, i know this is a lot of work so you really are appreciated, and i don't want to ever come off pushy, because i know at the end of the day this is something you totally don't have to do. This is something really special that you created, and i hope you get recognition in the GitHub community as one of the best programmers.
